lib/Haskell/Law/Num/Nat.agda

```diff
@@ -0,0 +1,37 @@
+module Haskell.Law.Num.Nat where
+
+open import Haskell.Prim
+open import Haskell.Prim.Num
+
+open import Haskell.Law.Equality
+open import Haskell.Law.Num.Def
+
+-- Where to put this?
```

Comment:
I think it's fine to put it here until we find a better place for it.

Reply:
Ok, then I will also gather all the other definitions here for now.